awstesting/integration/performance/s3DownloadManager/main_test.go

```diff
@@ -19,78 +21,143 @@ import (
 	flagSet.StringVar(&b.bucket, "bucket", "", "Bucket to use for benchmark")
 	flagSet.StringVar(&b.tempdir, "temp", os.TempDir(), "location to create temporary files")
+	flagSet.StringVar(&b.sizes, "size", fmt.Sprintf("%d,%d", 5*sdkio.MebiByte, 1*sdkio.GibiByte), "file sizes to benchmark separated by comma")
+	flagSet.StringVar(&b.parts, "part", fmt.Sprintf("%d,%d,%d", s3manager.DefaultDownloadPartSize, 25*sdkio.MebiByte, 100*sdkio.MebiByte), "part sizes to benchmark separated by comma")
+	flagSet.StringVar(&b.concurrency, "concurrency", fmt.Sprintf("%d,%d,%d", s3manager.DefaultDownloadConcurrency, 2*s3manager.DefaultDownloadConcurrency, 100), "part sizes to benchmark separated comma")
+	flagSet.StringVar(&b.bufferSize, "buffer", fmt.Sprintf("%d,%d", 0, 1*sdkio.MebiByte), "part sizes to benchmark separated comma")
+	flagSet.Int64Var(&b.uploadPartSize, "upload-part-size", 0, "upload part size, defaults to download part size if not specified")
 	b.clientConfig.SetupFlags(prefix, flagSet)
 }
 
-var benchStrategies = map[string]s3manager.WriterReadFromProvider{
-	"Unbuffered": nil,
-	"Buffered":   s3manager.NewPooledBufferedWriterReadFromProvider(int(sdkio.MebiByte)),
+func (b *BenchmarkConfig) BufferSizes() []int {
+	ints, err := b.stringToInt(b.bufferSize)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse file sizes: %v", err))
+	}
+
+	return ints
 }
 
-func BenchmarkDownload(b *testing.B) {
-	baseSdkConfig := SDKConfig{}
+func (b *BenchmarkConfig) FileSizes() []int64 {
+	ints, err := b.stringToInt64(b.sizes)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse file sizes: %v", err))
+	}
 
-	// FileSizes: 5 MB, 1 GB
-	for _, fileSize := range []int64{5 * sdkio.MebiByte, 1 * sdkio.GibiByte} {
-		key, err := setupDownloadTest(benchConfig.bucket, fileSize)
-		if err != nil {
-			b.Fatalf("failed to setup download test: %v", err)
-		}
-		f, err := ioutil.TempFile(benchConfig.tempdir, "BenchmarkDownload")
+	return ints
+}
+
+func (b *BenchmarkConfig) PartSizes() []int64 {
+	ints, err := b.stringToInt64(b.parts)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse part sizes: %v", err))
+	}
+
+	return ints
+}
+
+func (b *BenchmarkConfig) Concurrences() []int {
+	ints, err := b.stringToInt(b.concurrency)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse part sizes: %v", err))
+	}
+
+	return ints
+}
+
+func (b *BenchmarkConfig) stringToInt(s string) ([]int, error) {
+	int64s, err := b.stringToInt64(s)
+	if err != nil {
+		return nil, err
+	}
+
+	var ints []int
+	for i := range int64s {
+		ints = append(ints, int(int64s[i]))
+	}
+
+	return ints, nil
+}
+
+func (b *BenchmarkConfig) stringToInt64(s string) ([]int64, error) {
+	var sizes []int64
+
+	split := strings.Split(s, ",")
+
+	for _, size := range split {
+		i, err := strconv.ParseInt(strings.Trim(size, " "), 10, 64)
 		if err != nil {
-			b.Fatalf("failed to create temporary file: %v", err)
+			return nil, fmt.Errorf("invalid integer: %v", err)
```

Comment:
This error should probably include what `size` was, and maybe the original `s` string value.